mesonbuild/ast/introspection.py

```diff
@@ -137,17 +138,24 @@ def do_subproject(self, dirname: str) -> None:
             return
 
     def func_add_languages(self, node: BaseNode, args: T.List[TYPE_nvar], kwargs: T.Dict[str, TYPE_nvar]) -> None:
-        args = self.flatten_args(args)
-        for for_machine in [MachineChoice.BUILD, MachineChoice.HOST]:
-            for lang in sorted(args, key=compilers.sort_clink):
-                if isinstance(lang, StringNode):
-                    assert isinstance(lang.value, str)
-                    lang = lang.value
-                if not isinstance(lang, str):
-                    continue
-                lang = lang.lower()
-                if lang not in self.coredata.compilers[for_machine]:
-                    self.environment.detect_compiler_for(lang, for_machine)
+        if 'native' in kwargs:
+            native = kwargs.get('native', False)
```

Comment:
You should `flatten_kwargs` the `kwargs` first.

Reply:
Done

Reply:
... and it looks like it was never working correctly due to that, so I had the expected test result upside-down?